src/pages/signin.tsx

```diff
@@ -25,7 +27,7 @@ export default function Signin() {
   const {
     register,
     handleSubmit,
-    formState: { errors, isSubmitting, touchedFields },
+    formState: { errors, isSubmitting, touchedFields, isValid },
```

Comment:
```suggestion
    formState: { errors, isSubmitting, touchedFields, isValid, isDirty },
```

Maybe you can make use of the `isDirty` so the button won't be invalid from the start